compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -194,18 +194,47 @@ fn make_invalid_casting_error<'a, 'tcx>(
     )
 }
 
+/// If a cast from `from_ty` to `to_ty` is valid, returns an Ok containing the kind of
+/// the cast. In certain cases, including some invalid casts from array references
+/// to pointers, this may cause additional errors to be emitted and/or ICE error
```

Comment:
```suggestion
/// If a cast from `from_ty` to `to_ty` is valid, returns a `Some` containing the kind
/// of the cast. In certain cases, including some invalid casts from array references
/// to pointers, this may cause additional errors to be emitted and/or ICE error
```

This returns an `Option`, not a `Result`.